tokio/src/runtime/global.rs

```diff
@@ -30,26 +30,9 @@ where
     T: Future + Send + 'static,
     T::Output: Send + 'static,
 {
-    EXECUTOR.with(|current_executor| match current_executor.get() {
```

Comment:
I'm realizing now this might be an issue. We previously just dereferenced the executor pointer to spawn but now we need to clone + bump the reference count.

Reply:
Yeah, can we bundle this enum check in the spawn_handle.spawn possibly?

Reply:
I've removed this enum and module entirely.